src/main/java/tools/jackson/dataformat/xml/JacksonXmlAnnotationIntrospector.java

```diff
@@ -208,6 +215,13 @@ public PropertyName findNameForDeserialization(MapperConfig<?> config, Annotated
         PropertyName pn = PropertyName.merge(_findXmlName(a),
                 super.findNameForDeserialization(config, a));
         if (pn == null) {
+            JacksonXmlText jacksonXmlTextAnnotation = _findAnnotation(a, JacksonXmlText.class);
+
+            if (jacksonXmlTextAnnotation != null && jacksonXmlTextAnnotation.value() &&
+                !_cfgIntrospectorConfig.inferXmlTextPropertyName()) {
+                return _cfgIntrospectorConfig.xmlTextPropertyName();
+            }
+
             if (_hasOneOf(a, ANNOTATIONS_TO_INFER_XML_PROP)) {
```

Comment:
This will look for `@JacksonXmlText` OR `@JacksonXmlElementWrapper` and handling will now differ -- former should use configured "text element name", latter should not (should return "use default").

Reply:
I do not quite understand, the highlighted line only looks for `@JacksonXmlText`

Reply:
Sorry, I meant line 225 right after (`if (_hasOneOf(...)`).